tests/serialization/test_results.py

```diff
@@ -150,3 +150,41 @@ class MyHandler(ResultHandler):
     assert isinstance(new_result, results.ResultHandlerResult)
     assert new_result.location == "foo"
     assert new_result.value is None
+
+
+def test_custom_result():
+    class CustomResult(Result):
```

Comment:
Can you change the name of this custom class for the sake of the test? It's not clear from the test if we expect the class name to be stored in the `"type"` field or not (spoiler alert: it won't be)

Reply:
Good call